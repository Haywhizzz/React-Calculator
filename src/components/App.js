import Display from './Display';
import ButtonPanel from './ButtonPanel';

const App = () => (
  <>
    <h1>Calculator</h1>
    <Display result="0" />
    <ButtonPanel />
  </>
);

export default App;
