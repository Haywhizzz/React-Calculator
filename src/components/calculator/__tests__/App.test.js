import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';

it('App snapshot test', () => {
  const tree = renderer
    .create(<App />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
