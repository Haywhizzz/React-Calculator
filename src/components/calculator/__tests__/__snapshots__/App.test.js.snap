// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot test 1`] = `
<div
  id="calculator"
>
  <h1>
    Let's do some maths!
  </h1>
  <div
    id="display"
  >
    0
  </div>
  <div>
    <div>
      <button
        className="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "lightgray",
            "width": "25%",
          }
        }
        type="button"
      >
        AC
      </button>
      <button
        className="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "lightgray",
            "width": "25%",
          }
        }
        type="button"
      >
        +/-
      </button>
      <button
        className="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "lightgray",
            "width": "25%",
          }
        }
        type="button"
      >
        %
      </button>
      <button
        className="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "orange",
            "width": "25%",
          }
        }
        type="button"
      >
        รท
      </button>
    </div>
    <div>
      <button
        className="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "lightgray",
            "width": "25%",
          }
        }
        type="button"
      >
        7
      </button>
      <button
        className="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "lightgray",
            "width": "25%",
          }
        }
        type="button"
      >
        8
      </button>
      <button
        className="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "lightgray",
            "width": "25%",
          }
        }
        type="button"
      >
        9
      </button>
      <button
        className="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "orange",
            "width": "25%",
          }
        }
        type="button"
      >
        x
      </button>
    </div>
    <div>
      <button
        className="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "lightgray",
            "width": "25%",
          }
        }
        type="button"
      >
        4
      </button>
      <button
        className="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "lightgray",
            "width": "25%",
          }
        }
        type="button"
      >
        5
      </button>
      <button
        className="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "lightgray",
            "width": "25%",
          }
        }
        type="button"
      >
        6
      </button>
      <button
        className="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "orange",
            "width": "25%",
          }
        }
        type="button"
      >
        -
      </button>
    </div>
    <div>
      <button
        className="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "lightgray",
            "width": "25%",
          }
        }
        type="button"
      >
        1
      </button>
      <button
        className="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "lightgray",
            "width": "25%",
          }
        }
        type="button"
      >
        2
      </button>
      <button
        className="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "lightgray",
            "width": "25%",
          }
        }
        type="button"
      >
        3
      </button>
      <button
        className="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "orange",
            "width": "25%",
          }
        }
        type="button"
      >
        +
      </button>
    </div>
    <div>
      <button
        className="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "lightgray",
            "width": "50%",
          }
        }
        type="button"
      >
        0
      </button>
      <button
        className="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "lightgray",
            "width": "25%",
          }
        }
        type="button"
      >
        .
      </button>
      <button
        className="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "orange",
            "width": "25%",
          }
        }
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
